container/shim/src/modules/log_ingestor.js

```diff
@@ -41,155 +43,160 @@ const NGINX_LOG_KEYS_MAP = {
-        numBytesSent,
-        requestId,
-        localTime,
-        status,
-        requestDuration,
-        range,
-        cacheHit,
-        referrer,
-        userAgent,
-        http3,
-        httpProtocol,
-        url,
-      } = vars;
-
-      const cid = urlObj.pathname.split("/")[2];
-
-      if (cid === TESTING_CID) continue;
-
-      pending.push({
-        cacheHit,
-        clientAddress,
-        localTime,
-        numBytesSent,
-        range,
-        referrer,
-        requestDuration,
-        requestId,
-        userAgent,
-        httpStatusCode: status,
-        // If/when "httpProtocol" eventually contains HTTP/3.0, then
-        // the "http3" key can be removed.
-        httpProtocol: http3 || httpProtocol,
-        url,
-      });
-
-      valid++;
-      if (cacheHit) hits++;
-    }
-    if (valid > 0) {
-      debug(
-        `Parsed ${valid} valid retrievals in ${prettyBytes(read.length)} with hit rate of ${Number(
-          ((hits / valid) * 100).toFixed(2)
-        )}%`
-      );
-    }
-  } else {
-    if (hasRead) {
-      await fh.truncate();
-      await fh.close();
-      hasRead = false;
-      fh = await openFileHandle();
-    }
-  }
-  parseLogsTimer = setTimeout(parseLogs, Math.max(10_000 - valid, 1000));
-}
+  if (await isFileAccessible(LOG_FILE)) {
+    // stream the log file and parse the lines
+    const read = await readlines(LOG_FILE);
 
-async function openFileHandle() {
-  return await fsPromises.open("/usr/src/app/shared/nginx_log/node-access.log", "r+");
-}
+    if (read.lines.length) {
+      const logs = [];
+
+      for (let i = 0; i < read.lines.length; i++) {
+        // skip empty lines
+        if (read.lines[i] === "") continue;
+
+        // parse the line into an object
+        const parsed = parseSingleLine(read.lines[i]);
 
-export async function submitRetrievals() {
-  clearTimeout(submitRetrievalsTimer);
-  const length = pending.length;
-  if (length > 0) {
-    const body = {
-      nodeId,
-      filAddress: FIL_WALLET_ADDRESS,
-      bandwidthLogs: pending,
-    };
-    pending = [];
-    try {
-      debug(`Submitting ${length} pending retrievals`);
-      const startTime = Date.now();
-      await fetch(LOG_INGESTOR_URL, {
-        method: "POST",
-        body: JSON.stringify(body),
-        headers: {
-          Authentication: nodeToken,
-          "Content-Type": "application/json",
-        },
-      });
-      debug(`Submitted ${length} retrievals to wallet ${FIL_WALLET_ADDRESS} in ${Date.now() - startTime}ms`);
-    } catch (err) {
-      debug(`Failed to submit pending retrievals ${err.name} ${err.message}`);
-      pending = body.bandwidthLogs.concat(pending);
+        // add parsed log line if it is valid (only valid retrieval)
+        if (parsed) logs.push(parsed);
+      }
+
+      // submit the logs to the log ingestor
+      if (logs.length) {
+        try {
+          // submit parsed logs to the orchestrator
+          await submitLogs(logs);
```

Comment:
These comments are unnecessary imo. Sometimes code can actually be self documenting. I'm singling out this section but I mean it for the entire PR.

I like comments when they clarify business logic or add additional context. 
I dont like comments when they restate the code, it doesn't add value imo.

So for example instead of 

```js
        // skip empty lines
        if (read.lines[i] === "") continue;
```

you can do
```js
        const isEmptyLine = read.lines[i] === ""
        if (isEmptyLine) continue;
```

and it reads just as well. Except now, if that line of code ever changes, you don't have to update its corresponding comment.

I approve of this PR in its current state, but in the future I'd like to see less comments and more descriptive code.